sdk/communication/azure-communication-chat/azure/communication/chat/_shared/models.py

```diff
@@ -6,210 +6,150 @@
 
 from enum import Enum, EnumMeta
 from six import with_metaclass
+from typing import Mapping, Optional, Union, Any
+try:
+    from typing import Protocol, TypedDict
+except ImportError:
+    from typing_extensions import Protocol, TypedDict
 
-import msrest
+from azure.core import CaseInsensitiveEnumMeta
 
-class CommunicationError(msrest.serialization.Model):
-    """The Communication Services error.
 
-    Variables are only populated by the server, and will be ignored when sending a request.
+class CommunicationIdentifierKind(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
+    """Communication Identifier Kind.
+    """
+    UNKOWN = "unknown"
+    COMMUNICATION_USER = "communication_user"
+    PHONE_NUMBER = "phone_number"
+    MICROSOFT_TEAMS_USER = "microsoft_teams_user"
 
-    All required parameters must be populated in order to send to Azure.
 
-    :param code: Required. The error code.
-    :type code: str
-    :param message: Required. The error message.
-    :type message: str
-    :ivar target: The error target.
-    :vartype target: str
-    :ivar details: Further details about specific errors that led to this error.
-    :vartype details: list[~communication.models.CommunicationError]
-    :ivar inner_error: The inner error if any.
-    :vartype inner_error: ~communication.models.CommunicationError
+class CommunicationCloudEnvironment(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
     """
+    The cloud enviornment that the identifier belongs to
+    """
+
+    Public = "PUBLIC"
+    Dod = "DOD"
+    Gcch = "GCCH"
+
 
-    _validation = {
-        'code': {'required': True},
-        'message': {'required': True},
-        'target': {'readonly': True},
-        'details': {'readonly': True},
-        'inner_error': {'readonly': True},
-    }
-
-    _attribute_map = {
-        'code': {'key': 'code', 'type': 'str'},
-        'message': {'key': 'message', 'type': 'str'},
-        'target': {'key': 'target', 'type': 'str'},
-        'details': {'key': 'details', 'type': '[CommunicationError]'},
-        'inner_error': {'key': 'innererror', 'type': 'CommunicationError'},
-    }
-
-    def __init__(
-        self,
-        **kwargs
-    ):
-        super(CommunicationError, self).__init__(**kwargs)
-        self.code = kwargs['code']
-        self.message = kwargs['message']
-        self.target = None
-        self.details = None
-        self.inner_error = None
-
-
-class CommunicationErrorResponse(msrest.serialization.Model):
-    """The Communication Services error.
-
-    All required parameters must be populated in order to send to Azure.
-
-    :param error: Required. The Communication Services error.
-    :type error: ~communication.models.CommunicationError
+class CommunicationIdentifier(Protocol):
     """
+    Communication Identifier.
+
+    :ivar str id: The identifier.
```

Comment:
I've renamed to raw_id.
I would prefer to keep it on the Protocol - as it's the fundamental mechanism for these objects being somewhat "future-proof". It also seems that it would be necessary on `UnknownIdentifier`....